src/store/embalm/reducer.ts

```diff
@@ -196,46 +227,67 @@ export function embalmReducer(state: EmbalmState, action: Actions): EmbalmState
     case ActionType.SetArchAddressSearch:
       return { ...state, archAddressSearch: action.payload.search };
 
-    case ActionType.SetPayloadTxId:
-      return { ...state, payloadTxId: action.payload.txId };
-
-    case ActionType.SetShardsTxId:
-      return { ...state, shardsTxId: action.payload.txId };
-
     case ActionType.SetArchaeologistFullPeerId:
       return updateArchProperty(
         state,
-        action.payload.peerId.toString(),
-        'fullPeerId',
-        action.payload.peerId
-      );
+        action.payload.peerId.toString(), {
+        key: 'fullPeerId',
+        value: action.payload.peerId
+      });
 
     case ActionType.SetArchaeologistOnlineStatus:
       return updateArchProperty(
         state,
-        action.payload.peerId.toString(),
-        'isOnline',
-        action.payload.isOnline
-      );
+        action.payload.peerId, {
+        key: 'isOnline',
+        value: action.payload.isOnline
+      });
 
     case ActionType.SetArchaeologistConnection:
       return updateArchProperty(
         state,
-        action.payload.peerId.toString(),
-        'connection',
-        action.payload.connection
-      );
+        action.payload.peerId, {
+        key: 'connection',
+        value: action.payload.connection,
+        updateSelected: true,
+      });
+
+    case ActionType.SetArchaeologistSignature:
+      return updateArchProperty(
+        state,
+        action.payload.peerId, {
+        key: 'signature',
+        value: action.payload.signature,
+        updateSelected: true,
```

Comment:
Instead of this, why not call a second dispatch action to select the arch? `dispatch(selectArchaeologist(archAddress))`